File: 0867-transpose-matrix/0867-transpose-matrix.cpp
class Solution {
public:
    vector<vector<int>> transpose(vector<vector<int>>& matrix) {
        vector<vector<int>> ans;
        for(int r = 0; r < matrix[0].size(); r++){
            vector<int> tmp;
            for(int c = 0; c < matrix.size(); c++){
                tmp.push_back(matrix[c][r]);
            }
            ans.push_back(tmp);
        }
        return ans;
    }
};

File: 0867-transpose-matrix/0867-transpose-matrix.cs
public class Solution {
    public int[][] Transpose(int[][] matrix) {
        int rows = matrix[0].Length;
        int columns = matrix.Length;

        int[][] newMatrix = new int[rows][];

        for(int i = 0; i < rows; i++){
            newMatrix[i] = new int[columns];
            for(int j = 0; j < columns; j++){
                newMatrix[i][j] = matrix[j][i];
            }
        }

        return newMatrix;
    }
}